src/main/flight/mixer.c

```diff
@@ -345,6 +344,59 @@ static void applyFlipOverAfterCrashModeToMotors(void)
         }
     }
 }
+#ifdef USE_RPM_LIMITER
+static float getAverageRPM(void) // Placeholder function
+{
+    float averageRPM = 0;
+    for (int i = 0; i < getMotorCount(); i++) {
+        averageRPM += getDshotTelemetry(i);
+    }
+    return (averageRPM / (getMotorCount() * motorConfig()->motorPoleCount)) * 20;
+}
+
+static bool isMotorSaturated(void) //Placeholder function
+{
+    for (int i = 0; i < getMotorCount(); i++) {
+        if (motor[i] >= motorConfig()->maxthrottle) {
+            return true;
+        }
+    }
+    return false;
+}
+
+static void applyRPMLimiter(void)
+{
+    if (mixerConfig()->rpm_limiter && motorConfig()->dev.useDshotTelemetry && ARMING_FLAG(ARMED)) {
+        float averageRPM = getAverageRPM();
+        float averageRPMSmoothed = pt1FilterApply(&mixerRuntime.averageRPMFilter, averageRPM);
+        float smoothedRPMError = averageRPMSmoothed - mixerRuntime.rpmLimiterRPMLimit;
+        // PID
+        float rpmLimiterP = smoothedRPMError * mixerRuntime.rpmLimiterPGain; //+ when overspped
+        mixerRuntime.rpmLimiterI += smoothedRPMError * mixerRuntime.rpmLimiterIGain;
+        mixerRuntime.rpmLimiterI = MAX(0.0f, mixerRuntime.rpmLimiterI);
+        float rpmDerivative = smoothedRPMError - mixerRuntime.rpmLimiterPreviousSmoothedRPMError;
+        float rpmLimiterD = rpmDerivative * mixerRuntime.rpmLimiterDGain; // + when quickly going overspeed
+        float pidOutput = rpmLimiterP + mixerRuntime.rpmLimiterI + rpmLimiterD;
+        // Throttle limit learning
+        if (smoothedRPMError > -10.0f && rcCommand[THROTTLE] < 1950) {
+            mixerRuntime.rpmLimiterExpectedThrottleLimit *= 1.0f - 0.0006f;
```

Comment:
0.0006f and 0.0004f should be somehow multiplied by the pidloop frequency, to make sure throttle limit changes with the same speed on all fcs/pidloops?